Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/SolverHandler.cs

```diff
@@ -109,6 +129,28 @@ public bool UpdateSolvers
             set { updateSolvers = value; }
         }
 
+        protected readonly List<Solver> solvers = new List<Solver>();
+
+        /// <summary>
+        /// List of solvers that this handler will manage and update
+        /// </summary>
+        public IReadOnlyCollection<Solver> Solvers
+        {
+            get
+            {
+                return solvers.AsReadOnly();
```

Comment:
> solvers.AsReadOnly(); [](start = 22, length = 22)

Just to check here - can you double check that this doesn't allocate a new list on each call? If not, you can also I thik just cast this to a read only type list, which will accomplish the same thing without the memory allocation.

Reply:
Yes, it's a readonly wrapper around the existing collection
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.asreadonly?view=netframework-4.8